channeldb/payments.go

```diff
@@ -108,11 +115,21 @@ func (r FailureReason) String() string {
 		return "timeout"
 	case FailureReasonNoRoute:
 		return "no_route"
+	case FailureReasonError:
+		return "error"
+	case FailureReasonIncorrectPaymentDetails:
+		return "incorrect_payment_details"
 	}
 
 	return "unknown"
 }
 
+// Ref returns a reference to the failure reason. This is useful in a nilable
+// context.
+func (r FailureReason) Ref() *FailureReason {
```

Comment:
Why this over just having the caller take the address of the struct they hold?

Reply:
This is a utility function to get a pointer to value defined by const.

Reply:
Removed now